gstaichi/runtime/program_impls/llvm/llvm_program.h

```diff
@@ -191,11 +191,29 @@ class LlvmProgramImpl : public ProgramImpl {
 
     int child_id = root->child_id(dense_parent);
 
+    // The maximum cell size bytes is used to handle memory alignment for dlpack usage
+    // When compute field offset, SNode doesn't take the alignment into account yet, we need to mannully handle it here
+    size_t max_cell_size_bytes = root->ch[child_id].get()->cell_size_bytes;
     for (int i = 0; i < child_id; ++i) {
       SNode *child = root->ch[i].get();
-      offset += child->cell_size_bytes * child->num_cells_per_container;
+      size_t child_cell_size_bytes = child->cell_size_bytes;
```

Comment:
I wonder if it's worth factorizing out the work at each iteration into a separate function? @duburcqa any opinions/thougths on this?